pkg/image/prune/imagepruner.go

```diff
@@ -53,12 +67,59 @@ type ImagePruner interface {
 
 // imagePruner implements ImagePruner.
 type imagePruner struct {
-	g         graph.Graph
-	algorithm pruneAlgorithm
+	g              graph.Graph
+	algorithm      pruneAlgorithm
+	registryPinger registryPinger
 }
 
 var _ ImagePruner = &imagePruner{}
 
+// registryPinger performs a health check against a registry.
+type registryPinger interface {
+	// ping performs a health check against registry.
+	ping(registry string) error
+}
+
+// defaultRegistryPinger implements registryPinger.
+type defaultRegistryPinger struct {
+}
+
+func (drp *defaultRegistryPinger) ping(registry string) error {
+	healthzCheck := func(proto, registry string) error {
+		healthzResponse, err := http.Get(fmt.Sprintf("%s://%s/healthz", proto, registry))
```

Comment:
A slight terminology update:

1. (OpenShift) Remove prunable images from each associated image stream's status.tags
1. (Registry) Remove repo layer links (e.g. `repositories/andy/simple-openshift-sinatra-sti/_layers/sha256/55633dfc9f5407744645fd4e6626340c474cddf3df6a897230f5865ffaced640/link`)
1. (Registry) Remove repo manifest signatures (`repositories/andy/simple-openshift-sinatra-sti/_manifests/revisions/sha256/0b1910b3cdd54cebbee1fb4375fde718e135a8bffbee16eb782e6e2d451d703d and below`)
1. (Registry) Remove layer blobs (`blobs/sha256/7f/7fd4877c567299151622da51a06ee6e367001bebb37a35756a41725aeb3bb29f`)
1. (Registry) Remove signature blobs - we might want to defer this until the big refactor
1. (OpenShift) Remove images

The order of the deletions in the registry can be changed if needed.